clang/include/clang/CConv/RewriteUtils.h

```diff
@@ -93,10 +93,13 @@ class FunctionDeclReplacement :
            && (RewriteReturn || RewriteParams));
   }
 
-  SourceRange getSourceRange() const override {
-    FunctionTypeLoc TypeLoc =
-        getBaseTypeLoc(Decl->getTypeSourceInfo()->getTypeLoc())
-        .getAs<clang::FunctionTypeLoc>();
+  SourceRange getSourceRange(SourceManager &SM) const override {
+    TypeSourceInfo *TSInfo = Decl->getTypeSourceInfo();
+    if (!TSInfo)
+      return SourceRange(Decl->getBeginLoc(),
+                         getFunctionDeclarationEnd(Decl, SM));
```

Comment:
Why will `TSInfo` be `nullptr` and what is this added conditional doing in response?

Reply:
Very much unsure why TypeSourceInfo can be null, but it happens. This change is fixing something wrong with the macro rewriting changes in a previous pull request that I noticed while debugging the more recent changes.

If TypeSourceInfo comes back null, none of the fancy rewriting logic I have works since it all relies on that object. The conditional just falls back to what was done before that previous set of changes.